packages/http-client/src/http-client.ts

```diff
@@ -311,7 +376,9 @@ function logRequestInformation(request: HttpRequestConfig) {
  * @param options - An [[HttpRequestOptions]] of the http request for configuring e.g., CSRF token delegation. By default, the SDK will not fetch the CSRF token.
  * @returns A promise resolving to an [[HttpResponse]].
  */
-export function executeHttpRequest<T extends HttpRequestConfigWithOrigin>(
+export function executeHttpRequestWithOrigin<
```

Comment:
[nth] In the API doc, consider this:
Instead of "When reaching conflicts, values with high priorities are chosen." Explicitly tell what is going to happen -> custom > requestConfig.

Reply:
[req] Check API doc for http => HTTP

Reply:
[req] Also here, make sure the docs about fetching by default are correct.

Reply:
> Also here, make sure the docs about fetching by default are correct.
Nice catch, this has been changed.